cirq/ops/linear_combinations.py

```diff
@@ -205,3 +207,124 @@ def extend(expansion: value.LinearDict[str],
             extended_expansion = extend(expansion, op.qubits, self.qubits)
             result += extended_expansion * coefficient
         return result
+
+
+def _is_linear_dict_of_unit_pauli_string(
+        linear_dict: value.LinearDict[UnitPauliStringT]) -> bool:
+    if not isinstance(linear_dict, value.LinearDict):
+        return False
+    for k in linear_dict.keys():
+        if not isinstance(k, frozenset):
+            return False
+        for qid, pauli in k:
+            if not isinstance(qid, raw_types.Qid):
+                return False
+            if not isinstance(pauli, pauli_gates.Pauli):
+                return False
+
+    return True
+
+@value.value_equality(approximate=True)
+class PauliSum:
```

Comment:
Why not inherit from LinearCombinationOfOperations?

Reply:
see #1682 

We need the PauliStrings, -- which would otherwise serve as the key in a LinearDict -- to own their own coefficient